.github/workflows/deploy.yml

```diff
@@ -0,0 +1,51 @@
+name: GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
```

Comment:
For writing to gh-pages branch right?

Reply:
Exactly ;) I got this example for the action repo: https://github.com/peaceiris/actions-gh-pages